src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptions.cs

```diff
@@ -54,11 +54,11 @@ protected internal override void Setup(FeatureConfigurationContext context)
             {
                 var distributorAddress = context.Settings.GetOrDefault<string>("LegacyDistributor.Address");
 
-                var subscriberAddress = distributorAddress ?? context.Settings.LocalAddress();
+                var subscriberAddress = distributorAddress ?? context.Transport.SharedQueue;
```

Comment:
I think at this stage we can name `SharedQueue` `MainQueue`. I named the settings key `SharedQueue` but I don't think it should appear under than name. `LocalQueue` sounds also good.